core/src/test/scala/org/apache/spark/deploy/history/ApplicationCacheSuite.scala

```diff
@@ -370,7 +370,9 @@ class ApplicationCacheSuite extends SparkFunSuite with Logging with MockitoSugar
 
     val request = mock[HttpServletRequest]
     when(request.getMethod()).thenReturn("GET")
-    when(request.getRequestURI()).thenReturn("http://localhost:18080/history/local-123/jobs/job/")
+    when(request.getRequestURI()).thenReturn("/history/local-123/jobs/job/")
+    when(request.getRequestURL())
+      .thenReturn(new StringBuffer("http://localhost:18080/history/local-123/jobs/job/"))
```

Comment:
I'm wondering if we need `new StringBuffer` here.

Reply:
getRequestURL() returns a StringBuffer unlike getRequestURI() that returns a plain String, so, yes, we need to provide StringBuffer type to thenReturn.

Reply:
Got it.